jgiven-core/src/main/java/com/tngtech/jgiven/impl/StageLifecycleManager.java

```diff
@@ -44,6 +49,7 @@ boolean allAfterStageMethodsHaveBeenExecuted() {
 
     void executeAfterStageMethods(boolean fakeExecution) throws Throwable {
         executeLifecycleMethods(afterStageRegister, fakeExecution);
+        checkGuaranteedStatesAreInitialized(); //before or after the other after stage methods?
```

Comment:
in wouldn't this functionality be more apropriate to live in the `ValueInjector`, where the actual states are read?